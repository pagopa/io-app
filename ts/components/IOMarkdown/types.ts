import { VSpacer } from "@pagopa/io-app-design-system";
import {
  TxtHeaderNode,
  TxtListNode,
  TxtListItemNode,
  TxtParagraphNode,
  TxtStrNode,
  ASTNodeTypes,
  TxtStrongNode,
  TxtEmphasisNode,
  TxtLinkNode,
  TxtImageNode,
  TxtBlockQuoteNode,
  TxtCodeBlockNode,
  TxtCodeNode,
  TxtBreakNode,
  TxtHtmlNode,
  TxtHorizontalRuleNode,
  AnyTxtNode
} from "@textlint/ast-node-types";
import { ReactNode, ComponentProps } from "react";

export type Renderer = (txtNode: AnyTxtNodeWithSpacer) => ReactNode;
export type RuleRenderer<P, F extends Renderer | void = void> = (
  param: P,
  renderer: F
) => ReactNode;
export type AnyTxtNodeWithSpacer =
  | AnyTxtNode
  | { type: "Spacer"; size: number; key: string };

type TxtNodeMap = {
  [ASTNodeTypes.Header]: TxtHeaderNode;
  [ASTNodeTypes.Paragraph]: TxtParagraphNode;
  [ASTNodeTypes.Str]: TxtStrNode;
  [ASTNodeTypes.Strong]: TxtStrongNode;
  [ASTNodeTypes.Emphasis]: TxtEmphasisNode;
  [ASTNodeTypes.Link]: TxtLinkNode;
  [ASTNodeTypes.List]: TxtListNode;
  [ASTNodeTypes.ListItem]: TxtListItemNode;
  [ASTNodeTypes.Image]: TxtImageNode;
  [ASTNodeTypes.BlockQuote]: TxtBlockQuoteNode;
  [ASTNodeTypes.CodeBlock]: TxtCodeBlockNode;
  [ASTNodeTypes.Code]: TxtCodeNode;
  [ASTNodeTypes.Break]: TxtBreakNode;
  [ASTNodeTypes.Html]: TxtHtmlNode;
  [ASTNodeTypes.HorizontalRule]: TxtHorizontalRuleNode;
};
type RendererMap = {
  [K in keyof TxtNodeMap]: RuleRenderer<TxtNodeMap[K], Renderer>;
} & {
  [ASTNodeTypes.Comment]: RuleRenderer<null>;
  Spacer: RuleRenderer<ComponentProps<typeof VSpacer> & { key: string }>;
};

export type IOMarkdownRenderRules = RendererMap;
