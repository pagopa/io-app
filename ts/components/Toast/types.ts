import { HapticFeedbackTypes } from "react-native-haptic-feedback";
import { IOIcons } from "@pagopa/io-app-design-system";

export type ToastVariant = "neutral" | "error" | "info" | "success" | "warning";

export type Toast = {
  message: string;
  variant?: ToastVariant;
  icon?: IOIcons;
  hapticFeedback?: keyof typeof HapticFeedbackTypes;
};

export type ToastOptions = Omit<Toast, "message">;
