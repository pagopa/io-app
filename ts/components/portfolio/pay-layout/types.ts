/* subtitlesLS and subtitle are mutually exclusive */
type WithSubtitle = {
  hasMainSubtitle: true;
  hasSubtitlesLR: false;
  subtitleText: string;
};

type WithoutSubtitles = {
  hasMainSubtitle: false;
  hasSubtitlesLR: false;
};

type WithSubtitlesLR = {
  hasSubtitlesLR: true;
  hasMainSubtitle: false;
  subtitleLeftText?: string;
  subtitleRightText?: string;
};

type WithTouchable = {
  hasTouchable: true;
  touchableContent: React.ReactElement<any>;
};

type WithoutTouchable = {
  hasTouchable: false;
};

export type TopContentSubtitle = WithSubtitle & WithoutTouchable;
export type TopContentSubtitlesLR = WithSubtitlesLR & WithoutTouchable;
export type TopContentTouchable = WithoutSubtitles & WithTouchable;

export type TopContentSubtitleTouchable = WithSubtitle & WithTouchable;
export type TopContentSubtitlesLRTouchable = WithSubtitlesLR & WithTouchable;
export type TopContentNone = WithoutSubtitles & WithoutTouchable;

export function topContentSubtitle(subtitleText: string): TopContentSubtitle {
  return {
    hasMainSubtitle: true,
    hasSubtitlesLR: false,
    hasTouchable: false,
    subtitleText
  };
}

export function topContentSubtitlesLR(
  subtitleLeftText?: string,
  subtitleRightText?: string
): TopContentSubtitlesLR {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: true,
    hasTouchable: false,
    subtitleLeftText,
    subtitleRightText
  };
}

export function topContentTouchable(
  touchableContent: React.ReactElement<any>
): TopContentTouchable {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: false,
    hasTouchable: true,
    touchableContent
  };
}

export function topContentSubtitleTouchable(
  touchableContent: React.ReactElement<any>,
  subtitleText: string
): TopContentSubtitleTouchable {
  return {
    hasMainSubtitle: true,
    hasSubtitlesLR: false,
    hasTouchable: true,
    touchableContent,
    subtitleText
  };
}

export function topContentSubtitlesLRTouchable(
  touchableContent: React.ReactElement<any>,
  subtitleLeftText?: string,
  subtitleRightText?: string
): TopContentSubtitlesLRTouchable {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: true,
    hasTouchable: true,
    touchableContent,
    subtitleLeftText,
    subtitleRightText
  };
}

export function topContentNone(): TopContentNone {
  return {
    hasMainSubtitle: false,
    hasSubtitlesLR: false,
    hasTouchable: false
  };
}

export type TopContent =
  | TopContentSubtitle
  | TopContentSubtitleTouchable
  | TopContentSubtitlesLR
  | TopContentSubtitlesLRTouchable
  | TopContentTouchable
  | TopContentNone;
