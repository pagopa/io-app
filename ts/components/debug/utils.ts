type Primitive = string | number | boolean | null | undefined;

type TruncatableValue = Primitive | TruncatableObject | TruncatableArray;

interface TruncatableObject {
  [key: string]: TruncatableValue;
}

type TruncatableArray = Array<TruncatableValue>;

/**
 * Truncates all string values in an object or array structure to a specified maximum length.
 * This function creates a deep copy of the input, ensuring the original is not modified.
 *
 * @template T - The type of the input value, must extend TruncatableValue
 * @param {T} value - The value to process. Can be a string, number, boolean, null, undefined, array, or object
 * @param {number} [maxLength=250] - The maximum length for string values. Defaults to 250
 * @returns {T} A new value of the same type as the input, with all strings truncated to the specified length
 *
 * @example
 * const obj = { name: "Very long name...", age: 30, details: { bio: "Long bio..." } };
 * const truncated = truncateObjectStrings(obj, 10);
 * // Result: { name: "Very long...", age: 30, details: { bio: "Long bio..." } }
 */
export const truncateObjectStrings = <T extends TruncatableValue>(
  value: T,
  maxLength: number = 250
): T => {
  if (typeof value === "string") {
    return (
      value.length > maxLength ? value.slice(0, maxLength) + "..." : value
    ) as T;
  }

  if (Array.isArray(value)) {
    return value.map(item => truncateObjectStrings(item, maxLength)) as T;
  }

  if (typeof value === "object" && value !== null) {
    return Object.entries(value).reduce(
      (acc, [key, val]) => ({
        ...acc,
        [key]: truncateObjectStrings(val, maxLength)
      }),
      {}
    ) as T;
  }

  return value;
};
