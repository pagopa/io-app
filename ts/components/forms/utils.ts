/**
 * Useful methods to interact with redux-form
 *
 * @flow
 */

import * as React from 'react'
import { FieldProps } from 'redux-form'
import { Item, Input, View, Text } from 'native-base'
import { isEmail } from 'validator'

import I18n from '../../i18n'

const getValidatorMessage = (validatorId: string): string => {
  return I18n.t(`forms.validators.${validatorId}`)
}

const required = (value: string): ?string =>
  value ? undefined : getValidatorMessage('required')

const email = (value: string): ?string =>
  value && !isEmail(value) ? getValidatorMessage('email') : undefined

/**
 * A utility function that return the translated value for a property of a form field.
 */
export const getTraslatedFormFieldPropertyValue = (
  formId: string
): (string => string => string) => (fieldId: string): (string => string) => (
  propertyId: string
): string => {
  return I18n.t(`forms.${formId}.fields.${fieldId}.${propertyId}`)
}

/**
 * Methods used to validate redux-form `Field` components.
 * All methods takes a string, which represents the field value, as argument and returns
 * a error message if the validation is not fulfilled.
 */
export const validators = {
  required,
  email
}

/**
 * This method is used by redux-form `Field` components.
 * It takes as input the field properties and return a native-base `Input`.
 */
export const renderNativeBaseInput = ({
  input,
  meta: { touched, error, active },
  ...rest
}: FieldProps): React.ReactNode => {
  return (
    <View>
      <Item error={error && touched} active={active}>
        <Input {...input} {...rest} />
      </Item>
      {rest.showError && error && touched && <Text>{error}</Text>}
    </View>
  )
}
