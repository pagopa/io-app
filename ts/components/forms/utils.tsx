/**
 * Useful methods to interact with redux-form
 */

import { Input, Item, Text, View } from "native-base";
import * as React from "react";
import { WrappedFieldProps } from "redux-form";
import { isEmail } from "validator";

import I18n from "../../i18n";

const getValidatorMessage = (validatorId: string): string => {
  return I18n.t(`forms.validators.${validatorId}`);
};

const required = (value: string): string | undefined =>
  value ? undefined : getValidatorMessage("required");

const email = (value: string): string | undefined =>
  value && !isEmail(value) ? getValidatorMessage("email") : undefined;

/**
 * A utility function that return the translated value for a property of a form field.
 */
export const getTraslatedFormFieldPropertyValue = (
  formId: string
): ((_: string) => (_: string) => string) => (
  fieldId: string
): ((_: string) => string) => (propertyId: string): string => {
  return I18n.t(`forms.${formId}.fields.${fieldId}.${propertyId}`);
};

/**
 * Methods used to validate redux-form `Field` components.
 * All methods takes a string, which represents the field value, as argument and returns
 * a error message if the validation is not fulfilled.
 */
export const validators = {
  required,
  email
};

export interface NativeBaseInputProps {
  name?: string;
  placeholder?: string;
  showError?: boolean;
}

/**
 * This method is used by redux-form `Field` components.
 * It takes as input the field properties and return a native-base `Input`.
 */
export const renderNativeBaseInput: React.SFC<
  WrappedFieldProps & NativeBaseInputProps
> = ({ meta: { touched, error, active }, placeholder, showError }) => (
  <View>
    <Item error={error && touched} active={active}>
      <Input placeholder={placeholder} />
    </Item>
    {showError && error && touched && <Text>{error}</Text>}
  </View>
);
