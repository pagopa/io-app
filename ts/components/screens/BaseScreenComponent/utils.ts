import { BugReporting } from "instabug-reactnative";

import {
  DefaultReportAttachmentTypeConfiguration,
  TypeLogs,
  instabugLog,
  openInstabugQuestionReport,
  openInstabugReplies,
  setInstabugDeviceIdAttribute,
  setInstabugSupportTokenAttribute
} from "../../../boot/configureInstabug";
import { handleItemOnPress } from "../../../utils/url";
import {
  deriveCustomHandledLink,
  isIoInternalLink
} from "../../ui/Markdown/handlers/link";
import { getValueOrElse } from "../../../features/bonus/bpd/model/RemoteValue";
import { RequestAssistancePayload } from "../../ContextualHelp";

/**
 * Run side-effects from the Instabug library based on the type of support.
 */
export function handleOnContextualHelpDismissed(
  payload: RequestAssistancePayload,
  attachmentConfig: DefaultReportAttachmentTypeConfiguration
): void {
  const maybeSupportToken = getValueOrElse(payload.supportToken, undefined);
  const { supportType } = payload;

  switch (supportType) {
    case BugReporting.reportType.bug: {
      // Store/remove and log the support token only if is a new assistance request.
      // log on instabug the support token
      if (maybeSupportToken) {
        instabugLog(
          JSON.stringify(maybeSupportToken),
          TypeLogs.INFO,
          "support-token"
        );
      }
      // set or remove the properties
      setInstabugSupportTokenAttribute(maybeSupportToken);
      setInstabugDeviceIdAttribute(payload.deviceUniqueId);

      openInstabugQuestionReport(attachmentConfig);
      return;
    }

    case BugReporting.reportType.question: {
      openInstabugReplies();
      return;
    }

    default:
      return;
  }
}

export const handleOnLinkClicked = (hideHelp: () => void) => (url: string) => {
  // manage links with IO_INTERNAL_LINK_PREFIX as prefix
  if (isIoInternalLink(url)) {
    hideHelp();
    return;
  }

  // manage links with IO_CUSTOM_HANDLED_PRESS_PREFIX as prefix
  const customHandledLink = deriveCustomHandledLink(url);
  customHandledLink.map(link => handleItemOnPress(link.url)());
};
