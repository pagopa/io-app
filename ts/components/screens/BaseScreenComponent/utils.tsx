import { Millisecond } from "@pagopa/ts-commons/lib/units";
import * as E from "fp-ts/lib/Either";
import { pipe } from "fp-ts/lib/function";
import * as O from "fp-ts/lib/Option";
import React from "react";
import { ScreenCHData } from "../../../../definitions/content/ScreenCHData";
import { ContextualHelpData } from "../../../features/zendesk/screens/ZendeskSupportHelpCenter";
import I18n from "../../../i18n";
import { handleItemOnPress } from "../../../utils/url";
import Markdown from "../../ui/Markdown";
import {
  deriveCustomHandledLink,
  isIoInternalLink
} from "../../ui/Markdown/handlers/link";
import { ContextualHelpProps, ContextualHelpPropsMarkdown } from "./index";

export const handleOnLinkClicked = (hideHelp: () => void) => (url: string) => {
  // manage links with IO_INTERNAL_LINK_PREFIX as prefix
  if (isIoInternalLink(url)) {
    hideHelp();
    return;
  }

  // manage links with IO_CUSTOM_HANDLED_PRESS_PREFIX as prefix
  const customHandledLink = deriveCustomHandledLink(url);
  pipe(
    customHandledLink,
    E.map(link => handleItemOnPress(link.url)())
  );
};

/**
 * Extract a title and a body (ContextualHelpProps) if the contextualHelp or the contextualHelpMarkdown in input are defined,
 * otherwise returns undefined.
 * @param contextualHelp
 * @param contextualHelpMarkdown
 * @param onLoadEnd
 * @param onLinkClicked
 * @param shouldHandleLink
 */
export const getContextualHelpConfig = (
  contextualHelp: ContextualHelpProps | undefined,
  contextualHelpMarkdown: ContextualHelpPropsMarkdown | undefined,
  onLoadEnd: () => void,
  onLinkClicked: (url: string) => void,
  shouldHandleLink?: (url: string) => boolean
): ContextualHelpProps | undefined =>
  contextualHelp
    ? { body: contextualHelp.body, title: contextualHelp.title }
    : contextualHelpMarkdown
    ? {
        body: () => (
          <Markdown
            onLinkClicked={onLinkClicked}
            onLoadEnd={onLoadEnd}
            shouldHandleLink={shouldHandleLink}
          >
            {I18n.t(contextualHelpMarkdown.body)}
          </Markdown>
        ),
        title: I18n.t(contextualHelpMarkdown.title)
      }
    : undefined;

/**
 If contextualData (loaded from the content server) contains the route of the current screen,
 title, content, faqs are read from it, otherwise they came from the locales stored in app
 */
export const getContextualHelpData = (
  maybeContextualData: O.Option<ScreenCHData>,
  defaultData: ContextualHelpData,
  onReady: () => void
): ContextualHelpData =>
  pipe(
    maybeContextualData,
    O.fold(
      () => defaultData,
      data => ({
        title: data.title,
        content: <Markdown onLoadEnd={onReady}>{data.content}</Markdown>,
        faqs: pipe(
          data.faqs,
          O.fromNullable,
          O
            // ensure the array is defined and not empty
            .chainNullableK(faqs => (faqs.length > 0 ? faqs : undefined)),
          O
            // if remote faqs are not defined or empty, fallback to the local ones
            .fold(
              () => defaultData.faqs,
              fqs => fqs.map(f => ({ title: f.title, content: f.body }))
            )
        )
      })
    )
  );

export const reloadContextualHelpDataThreshold = (30 * 1000) as Millisecond;
