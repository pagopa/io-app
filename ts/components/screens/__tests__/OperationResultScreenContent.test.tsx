import React from "react";
import { createStore } from "redux";
import ROUTES from "../../../navigation/routes";
import { applicationChangeState } from "../../../store/actions/application";
import { appReducer } from "../../../store/reducers";
import { GlobalState } from "../../../store/reducers/types";
import { renderScreenWithNavigationStoreContext } from "../../../utils/testWrapper";
import {
  OperationResultScreenContent,
  OperationResultScreenContentProps
} from "../OperationResultScreenContent";

const defaultProps: OperationResultScreenContentProps = {
  title: "title",
  subtitle: "subtitle",
  action: {
    onPress: jest.fn(),
    label: "Action",
    accessibilityLabel: "Action"
  },
  secondaryAction: {
    onPress: jest.fn(),
    label: "Secondary Action",
    accessibilityLabel: "Secondary Action"
  }
};

describe("OperationResultScreenContent", () => {
  it("should match the snapshot with default props", () => {
    expect(renderComponent(defaultProps)).toMatchSnapshot();
  });
});

function renderComponent(
  props: React.ComponentProps<typeof OperationResultScreenContent>
) {
  const globalState = appReducer(undefined, applicationChangeState("active"));
  return renderScreenWithNavigationStoreContext<GlobalState>(
    () => <OperationResultScreenContent {...props} />,
    ROUTES.WALLET_HOME,
    {},
    createStore(appReducer, globalState as any)
  );
}
