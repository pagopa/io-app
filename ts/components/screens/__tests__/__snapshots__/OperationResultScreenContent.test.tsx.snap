// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OperationResultScreenContent should match the snapshot with default props 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          collapsable={false}
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                collapsable={false}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                WALLET_HOME
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            collapsable={false}
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={1}
              handlerType="PanGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "flexGrow": 1,
                          "marginHorizontal": 24,
                        }
                      }
                    >
                      <RCTScrollView
                        centerContent={true}
                        collapsable={false}
                        contentContainerStyle={
                          Array [
                            Object {
                              "alignContent": "center",
                              "alignItems": "stretch",
                              "flex": 1,
                              "justifyContent": "center",
                            },
                            false,
                          ]
                        }
                        handlerTag={2}
                        handlerType="NativeViewGestureHandler"
                        onGestureHandlerEvent={[Function]}
                        onGestureHandlerStateChange={[Function]}
                        waitFor={
                          Array [
                            Object {
                              "current": null,
                            },
                          ]
                        }
                      >
                        <View>
                          <Text
                            allowFontScaling={false}
                            color="bluegreyDark"
                            defaultColor="bluegreyDark"
                            defaultWeight="SemiBold"
                            font="TitilliumWeb"
                            fontStyle={
                              Object {
                                "fontSize": 24,
                                "lineHeight": 34,
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "textAlign": "center",
                                },
                                Object {
                                  "fontSize": 24,
                                  "lineHeight": 34,
                                },
                                Object {
                                  "color": "#17324D",
                                  "fontFamily": "Titillium Web",
                                  "fontStyle": "normal",
                                  "fontWeight": "600",
                                },
                              ]
                            }
                            weight="SemiBold"
                          >
                            title
                          </Text>
                          <View
                            style={
                              Object {
                                "height": 8,
                              }
                            }
                          />
                          <Text
                            color="grey-650"
                            defaultColor="blue"
                            defaultWeight="Bold"
                            font="TitilliumWeb"
                            fontStyle={
                              Object {
                                "fontSize": 14,
                                "lineHeight": 21,
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "textAlign": "center",
                                },
                                Object {
                                  "fontSize": 14,
                                  "lineHeight": 21,
                                },
                                Object {
                                  "color": "#636B82",
                                  "fontFamily": "Titillium Web",
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                              ]
                            }
                            weight="Regular"
                          >
                            subtitle
                          </Text>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 24,
                                }
                              }
                            />
                            <View>
                              <View
                                accessibilityLabel="Action"
                                accessibilityRole="button"
                                accessibilityState={
                                  Object {
                                    "busy": false,
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "flex-start",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "borderRadius": 4,
                                        "elevation": 0,
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "paddingHorizontal": 16,
                                        "textAlignVertical": "center",
                                      },
                                      Object {
                                        "overflow": "hidden",
                                      },
                                      false,
                                      Object {
                                        "height": 40,
                                      },
                                      Object {
                                        "backgroundColor": "#0073E6",
                                      },
                                      Object {
                                        "backgroundColor": undefined,
                                        "transform": Array [
                                          Object {
                                            "scale": undefined,
                                          },
                                        ],
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    entering={[Function]}
                                    exiting={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "alignItems": "center",
                                          "flexDirection": "row",
                                          "justifyContent": "center",
                                        },
                                        false,
                                      ]
                                    }
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      color="white"
                                      defaultColor="white"
                                      defaultWeight="Bold"
                                      ellipsizeMode="tail"
                                      font="TitilliumWeb"
                                      fontStyle={
                                        Object {
                                          "fontSize": 16,
                                        }
                                      }
                                      maxFontSizeMultiplier={1.3}
                                      numberOfLines={1}
                                      style={
                                        Array [
                                          Object {
                                            "alignSelf": "center",
                                          },
                                          Object {
                                            "fontSize": 16,
                                          },
                                          Object {
                                            "color": "#FFFFFF",
                                            "fontFamily": "Titillium Web",
                                            "fontStyle": "normal",
                                            "fontWeight": "700",
                                          },
                                        ]
                                      }
                                      weight="Bold"
                                    >
                                      Action
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 24,
                                }
                              }
                            />
                            <View>
                              <View
                                accessibilityLabel="Secondary Action"
                                accessibilityRole="button"
                                accessibilityState={
                                  Object {
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                hitSlop={
                                  Object {
                                    "bottom": 14,
                                    "left": 24,
                                    "right": 24,
                                    "top": 14,
                                  }
                                }
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                onTouchEnd={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "flex-start",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "elevation": 0,
                                        "flexDirection": "row",
                                        "justifyContent": "center",
                                        "textAlignVertical": "center",
                                      },
                                      false,
                                      Object {},
                                      Object {
                                        "transform": Array [
                                          Object {
                                            "scale": undefined,
                                          },
                                        ],
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    ellipsizeMode="tail"
                                    maxFontSizeMultiplier={1.3}
                                    numberOfLines={1}
                                    style={
                                      Array [
                                        Object {
                                          "fontFamily": "Titillium Web",
                                          "fontSize": 16,
                                          "fontStyle": "normal",
                                          "fontWeight": "700",
                                        },
                                        Object {
                                          "color": "#0073E6",
                                        },
                                        Object {
                                          "color": undefined,
                                        },
                                      ]
                                    }
                                  >
                                    Secondary Action
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
